### PR TITLE
Add Recursion Folder and Tower of Hanoi algorithm 

### PR DESCRIPTION
**Pull request by knotman90**
It also contains code for pretty printing the status of the tower.
refresh speed is controlled via sleeptime variable.

### CODE REVIEW

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp]: Since the number of piles are fixed (3), declare `piles` as:
```C++
array<vector<int>, 3> piles;
```

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp]: Please remove `ndisks` as a global variable. Also `sleeptime` can be removed as there is no need for the sleep functionality.

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp]: No need of `std::`

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp]: You can change this to:
```C++
int num_discs;
cout << "Enter the number of discs : ";
cin >> num_discs;
```

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp]: `usleep()` can be removed as there is no need for the sleep functionality.

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp]: Can this be simplified? Also please add a comment to explain how/why a pile is selected as a support.

**Review by faheel — CHANGES_REQUESTED** [review]
Sorry for the late review :sweat_smile:

File: Recursion/TowerHanoi.cpp
Comment (knotman90) [reply Recursion/TowerHanoi.cpp]: If there is no need for usleep() then I think I can get rid of the drawHanoi function too. It would be unusable withouth a sleep cause it will just execute too fast.

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp:35]: I think that using a solver class is unnecessary. It was fine the way it was before:
- A function that recursively solves the Tower of Hanoi problem, and after each move calls:
- Another function that draws the current status of the 3 piles.

This way it's much simple to understand. The initialisation etc. can be done in `main()`.

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp:17]: `(not considering the internal stack space used for recursion)`

File: Recursion/TowerHanoi.cpp
Comment (faheel) [inline Recursion/TowerHanoi.cpp:16]: It would be better if you used uppercase variables while mentioning the time and space complexity as they are easier to spot while reading.
For eg., consider:
`... where n is the number ...`
vs
`... where N is the number ...`
Not much of a difference, but that's how it's written elsewhere in the repo.

**Review by faheel — CHANGES_REQUESTED** [review]
Please use 4 spaces as one level of indentation.

**Comment by faheel** [comment]
There have been lots of suggestions recently by interested contributors to add more algorithms and data structures to the project, which I have currently put on hold because I want to close issue #74 first, as it would require a complete reorganisation of files and some refactoring of code.

I previously thought that I would work on #74 once the currently active PRs are merged, but I feel that it would take much longer that way. So, if it's not too inconvenient for you, I would like to work on issue #74 as soon as possible. The remaining work on this PR (and others) can be carried on afterwards. Once the project is refactored, all you would need to do is `git fetch upstream` and move the code that you wrote to the new appropriate directory.

So, what do you say?

**Comment by knotman90** [comment]
Sounds good to me =)

**Comment by faheel** [comment]
Closing this due to lack of activity for a long time.

**Pull request closed without merging**
